Assets/Scripts/Presenters/BuildingPresenter.cs

```diff
@@ -0,0 +1,122 @@
+                {(_gameManager.Cpu1, InventoryItems.Gold), new List<GameObject>() {bTemple}},
+                {(_gameManager.Cpu1, InventoryItems.Steel), new List<GameObject>() {bShelter}},
+                {(_gameManager.Cpu1, InventoryItems.Insulation), new List<GameObject> {bWarehouse}},
+                {(_gameManager.Cpu1, InventoryItems.Stone), new List<GameObject> {bWaveBreaker}},
+
+                {(_gameManager.Cpu2, InventoryItems.Wood), new List<GameObject>() {cObservationPoint}},
+                {(_gameManager.Cpu2, InventoryItems.Lenses), new List<GameObject>() {cObservatory}},
+                {(_gameManager.Cpu2, InventoryItems.Clay), new List<GameObject>() {cBrokenHomes, cRepairedHomes}},
+                {(_gameManager.Cpu2, InventoryItems.Gold), new List<GameObject>() {cTemple}},
+                {(_gameManager.Cpu2, InventoryItems.Steel), new List<GameObject>() {cShelter}},
+                {(_gameManager.Cpu2, InventoryItems.Insulation), new List<GameObject> {cWarehouse}},
+                {(_gameManager.Cpu2, InventoryItems.Stone), new List<GameObject> {cWaveBreaker}},
+            };
+
+            _buildStructures = new Dictionary<Tribe, List<GameObject>>
+            {
+                [_gameManager.Player] = new List<GameObject>(),
+                [_gameManager.Cpu1] = new List<GameObject>(),
+                [_gameManager.Cpu2] = new List<GameObject>()
+            };
+
+            _gameManager.Player.Inventory.InventoryUpdate += InventoryUpdateEventHandler;
+            _gameManager.Cpu1.Inventory.InventoryUpdate += InventoryUpdateEventHandler;
+            _gameManager.Cpu2.Inventory.InventoryUpdate += InventoryUpdateEventHandler;
+            
+            _tribes = new[] {_gameManager.Player, _gameManager.Cpu1, _gameManager.Cpu2};
+        }
+    
+        private void InventoryUpdateEventHandler(object sender, EventArgs eventArgs)
+        {
+            foreach (var tribe in _tribes)
+            {
+                var resource = _buildingService?.CheckIfBuildingPossible(tribe);
+                if (resource == null) continue;
+                PlaceBuilding(tribe, (InventoryItems) resource);
+                _buildingService.AddBuildingPoints(_tribes, (InventoryItems) resource, tribe);
+                _buildingService.RemoveBuildingResourcesFromInventory(tribe,(InventoryItems) resource);
+            }
+        }
+        
+        private void PlaceBuilding(Tribe tribe, InventoryItems resource)
+        {
+            var modelList = _buildableStructures[(tribe, resource)];
+
+            if (modelList.Count == 1)
+            {
+                _buildStructures[tribe].Add(modelList[0]);
+                modelList[0].SetActive(true);
+            }
+
+            if (modelList.Count == 2)
+            {
+                _buildStructures[tribe].Add(modelList[1]);
+                modelList[0].SetActive(false);
+                modelList[1].SetActive(true);
+            }

```

Comment:
Add comments explaining what this does please

Reply:
Yea, if you could add one please, after that its ready for merch imo